### ISSUE TITLE
Breaking change in DocumentClient constructor in v1.15.0

### ISSUE THREAD

**Issue by bcrobinson** [Azure/azure-documentdb-dotnet#307]
The release notes at [docs.microsoft.com](https://docs.microsoft.com/en-gb/azure/cosmos-db/documentdb-sdk-dotnet) mention that support for custom JsonSerializerSettings has been added  when instanciating a DocumentClient.

This has been done by adding an extra parameter to the existing constructors for JsonSerializerSettings with a default value.
```csharp
// V 1.14.1
public DocumentClient(Uri serviceEndpoint, SecureString authKey, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null)
public DocumentClient(Uri serviceEndpoint, string authKeyOrResourceToken, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null)
public DocumentClient(Uri serviceEndpoint, IList<Permission> permissionFeed, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null)

// V 1.15.0
public DocumentClient(Uri serviceEndpoint, SecureString authKey, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null, JsonSerializerSettings serializerSettings = null)
public DocumentClient(Uri serviceEndpoint, string authKeyOrResourceToken, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null, JsonSerializerSettings serializerSettings = null)
public DocumentClient(Uri serviceEndpoint, IList<Permission> permissionFeed, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null) : this(serviceEndpoint, DocumentClient.GetResourceTokens(permissionFeed), connectionPolicy, desiredConsistencyLevel)
```

While this is not a breaking change for code compiled against the new version, we have a dependency built against v1.14.1 and using this in a project referencing v1.15.0. 
This breaks at runtime with the following exception:
```
Method not found: 'Void Microsoft.Azure.Documents.Client.DocumentClient..ctor(System.Uri, System.String, Microsoft.Azure.Documents.Client.ConnectionPolicy, System.Nullable`1<Microsoft.Azure.Documents.ConsistencyLevel>)'.
```

Unfortunately this leaves the sdk a bit stuck, as to keep backwards compatibility a two constructors are needed, one with the original parameters and one with the JsonSerializerSettings parameter.

But this would puts you in a situation where you break current users at compile time.
E.g.
```csharp
public DocumentClient(Uri serviceEndpoint, SecureString authKey, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null)
public DocumentClient(Uri serviceEndpoint, SecureString authKey, ConnectionPolicy connectionPolicy = null, ConsistencyLevel? desiredConsistencyLevel = null, JsonSerializerSettings serializerSettings = null)

// this will result in "The call is ambiguous" compile error as both ctors are valid.
var client1 = new DocumentClient(endpoint, key);
```

We can currently work around this by downgrading our dependency on the Microsoft.Azure.DocumentDB nuget package to v1.14.1 until the upstream dependency has been update to 1.15.0.

**Comment by DaveAurionix** [comment]
Yes, this stung us too.

**Comment by rnagpal** [comment]
We are working on a fix for that and the new update will be released early next week.

**Comment by DaveAurionix** [comment]
That's great, thank you.

**Comment by rnagpal** [comment]
This has been fixed in 1.16.0 release of the SDK.
